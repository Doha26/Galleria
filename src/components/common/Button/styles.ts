import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
    marginHorizontal: 8,
    marginVertical: 2,
    borderRadius: 30,
    height: 60,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
export default styles;
