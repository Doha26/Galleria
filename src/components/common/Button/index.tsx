import React from 'react';
import {
  View,
  Platform,
  TouchableOpacity,
  ActivityIndicator,
  TouchableWithoutFeedback,
} from 'react-native';
import PropTypes from 'prop-types';
import Text from '../Text/Text';
import colors from '~/theming/colors';
import styles from '~/components/common/Button/styles';

const propTypes = {
  loading: PropTypes.bool,
  color: PropTypes.string,
  disabled: PropTypes.bool,
  tintColor: PropTypes.string,
  transparent: PropTypes.bool,
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const defaultProps = {
  loading: false,
  disabled: false,
  transparent: false,
  color: colors.black,
  tintColor: colors.black,
  fontSize: 22,
};

const Button = ({
  text,
  color,
  loading,
  onPress,
  disabled,
  tintColor,
  transparent,
  fontSize,
}: {
  text: string;
  color: string;
  loading?: boolean;
  onPress: () => void;
  disabled?: boolean;
  tintColor?: string;
  transparent?: boolean;
  fontSize?: number;
}) => {
  const {container} = styles;
  const buttonContainerStyle = [container];
  let buttonContent;

  if (loading) {
    buttonContent = (
      <ActivityIndicator
        color={colors.white}
        style={{marginVertical: Platform.OS === 'ios' ? 10 : 0}}
        size={Platform.OS === 'ios' ? 1 : 24}
      />
    );
  } else {
    buttonContent = (
      <Text
        color={tintColor}
        style={{textAlign: 'center', color: tintColor, fontSize, fontWeight: 'bold'}}>
        {text}
      </Text>
    );
  }

  if (disabled || loading) {
    // @ts-ignore
    buttonContainerStyle.push({backgroundColor: colors.darkgray});

    return (
      <TouchableWithoutFeedback>
        <View style={buttonContainerStyle}>{buttonContent}</View>
      </TouchableWithoutFeedback>
    );
  }

  if (!transparent) {
    // @ts-ignore
    buttonContainerStyle.push({backgroundColor: color});
  }

  return (
    <TouchableOpacity onPress={onPress} style={buttonContainerStyle} activeOpacity={0.7}>
      <View>{buttonContent}</View>
    </TouchableOpacity>
  );
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
