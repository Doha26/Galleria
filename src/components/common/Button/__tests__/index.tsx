import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '~/components/common/Button';
import colors from '~/theming/colors';

describe('Button Component', () => {
  it('should render without issues', () => {
    const component = shallow(
      <Button text="Test Button" color={colors.black} onPress={() => {}} />,
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should be call onPress events', () => {
    const onPress = jest.fn();
    console.log = jest.fn();
    const wrapper = shallow(<Button text="Test Button" color={colors.black} onPress={() => {}} />);

    // Call default onPress
    // @ts-ignore
    wrapper.find('TouchableOpacity').first().props().onPress();

    // @ts-ignore
    expect(console.log.mock.calls[0][0]).toBe(`Please attach a method to this component`);

    wrapper.setProps({onPress});

    // Call our custom onPress
    // @ts-ignore
    wrapper.find('TouchableOpacity').first().props().onPress();

    expect(onPress).toHaveBeenCalled();
  });
});
