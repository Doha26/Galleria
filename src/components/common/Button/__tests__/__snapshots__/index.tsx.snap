// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render without issues 1`] = `
<Component
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 30,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 8,
        "marginVertical": 2,
        "paddingHorizontal": 20,
      },
      Object {
        "backgroundColor": "#050405",
      },
    ]
  }
>
  <Component>
    <Text
      color="#050405"
      extraLarge={false}
      large={false}
      numberOfLines={2}
      small={false}
      style={
        Object {
          "color": "#050405",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      theme={null}
    >
      Test Button
    </Text>
  </Component>
</Component>
`;
